xarray/core/dataset.py

```diff
@@ -6294,5 +6294,55 @@ def idxmax(
             ),
         )
 
+    def argmin(self, dim=None, axis=None, **kwargs):
+        if dim is None and axis is None:
+            warnings.warn(
+                "Behaviour of DataArray.argmin() with neither dim nor axis argument "
```

Comment:
This warning mentions DataArray, but this is on the Dataset method.

Reply:
That is intentional, although maybe it should mention both `DataArray.argmin()` and `Variable.argmin()` - the current behaviour of `DataArray.argmin()` with no `dim` argument is deprecated - it will change in future to return a `dict`. At that point it wouldn't make sense for `Dataset.argmin()` with no dim argument to return a result, because it doesn't make much sense to return a `dict` of `Dataset`s (and if the method returned a result using `reduce` to wrap `duck_array_ops.argmin`, it would return a single `Dataset` with the flattened index value for each member), so I think `Dataset.argmin()` with no `dim` argument should raise an error once the change is made to `DataArray.argmin()` and `Variable.argmin()`. I'll reword the message to make this clearer...